R/binning.R

```diff
@@ -0,0 +1,75 @@
+#' @title Binning
+#'
+#' @description
+#'
+#' Aggregate values in `x` for bins defined on `toBin`: all values
+#' in `x` for values in `toBin` falling into a bin (defined on `toBin`) are
+#' aggregated with the provided function `FUN`.
+#'
+#' @param x `numeric` with the values that should be aggregated/binned.
+#'
+#' @param toBin `numeric` with same length than `x` with values to be used for
+#'     the binning.
+#'
+#' @param binSize `numeric(1)` with the size of a bin.
+#'
+#' @param breaks `numeric` defining the breaks (bins).
+#'
+#' @param FUN `function` to be used to aggregate values of `x` falling into the
+#'     bins defined by `breaks`.
+#'
+#' @return `list` with elements `x` (aggregated values of `x`) and `mids` (the
+#'     bin mid points).
+#'
+#' @author Johannes Rainer, Sebastian Gibbals
```

Comment:
??? Gibbals??? WTF, must have been autocomplete on emacs . Sorry for that.

Reply:
Hm, and what if we define a method `bin` in `MsCoreUtils` (could eventually also move that later to `ProtGenerics`). I'm just afraid that it will bite us some day to have a function `bin` and a method `bin`...

OK with changing `binSize` to `size`. What about changing `toBin` to `y`?

Reply:
I am fine with `y`, too.

Reply:
Remains the choice of the name `bin` *vs* `binValues`...

Reply:
It's `bin`.